arguments.py

```diff
@@ -34,11 +55,11 @@ def parse_args() -> argparse.Namespace:
 
     args = parser.parse_args()
     if args.subcommand == "deploy":
-        args.steps = args.steps.split(",")
-        args.skip_steps = args.skip_steps.split(",")
+        args.steps = list(filter(None, args.steps.split(",")))
+        args.skip_steps = list(filter(None, args.skip_steps.split(",")))
 
         invalid_steps = [step for step in args.steps + args.skip_steps if step not in VALID_STEPS]
-        if set:
```

Comment:
`set` is a class in python. I would clean up your commits to not even introduce this. You are overwriting the class `set` with a var